client/driver/docker.go

```diff
@@ -884,6 +888,9 @@ func (d *DockerDriver) detectIP(c *docker.Container) (string, bool) {
 		}
 
 		ip = net.IPAddress
+		if d.driverConfig.UseIPv6Address {
+			ip = net.GlobalIPv6Address
```

Comment:
Hm, it seems like we probably want to set `auto = true` here as well. While this complicates the already unfortunately confusing `advertise = "auto"` logic, I think it will do what users expect by automatically advertising the routable address for the container when it's possible to discover it.

Perhaps a better question is: It should be exceedingly rare that someone wants `use_ipv6_address = true` and `advertise = "host"`, correct?

Reply:
(this has been addressed; leaving for historical purposes)